servers/zts/src/main/java/com/yahoo/athenz/zts/store/DataStore.java

```diff
@@ -1936,10 +1939,28 @@ public List<Role> getRolesByDomain(String domain) {
         return domainData.getRoles();
     }
 
+    @Override
+    public List<PublicKeyEntry> getPubKeysByService(String domain, String service) {
+        DomainData domainData = getDomainData(domain);
+        if (domainData == null) {
+            LOGGER.error("domainData can not be null, domain: {}, service: {}", domain, service);
+            return new ArrayList<>();
+        }
+
+        for (ServiceIdentity serviceIdentity: domainData.getServices()) {
+            if (serviceIdentity.getName().equalsIgnoreCase(domain + "." + service)) {
```

Comment:
we should move the "domain + "." + service" outside of the for block so we're not executing for every call in the iteration since the value does not change